src/sentinel.c

```diff
@@ -4955,7 +4973,7 @@ void sentinelFailoverWaitStart(sentinelRedisInstance *ri) {
     /* If I'm not the leader, and it is not a forced failover via
      * SENTINEL FAILOVER, then I can't continue with the failover. */
     if (!isleader && !(ri->flags & SRI_FORCE_FAILOVER)) {
-        int election_timeout = sentinel_election_timeout;
```

Comment:
@yossigo @hwware can you please assess the effects of this bug?

Reply:
Hey @oranagra , this is a potential bug assuming the value of sentinel_election_timeout is set to a value of greater than 2147483647. mstime_t is long long (defined in https://github.com/redis/redis/blob/f11a2d4dd764c996b2d0c0cb5abde13f2445b40c/src/server.h#L58). For example a value of mstime_t greater than INTMAX will be set as negative in the current case when . It makes sense to use mstime_t for election_timeout for consistency and to avoid the potential bug. But I don't know in what case the value for sentinel_election_timeout will be set to a high value greater than 2147483647.

Reply:
Hey @oranagra, this is a potential bug when the value of sentinel_election_timeout is larger than 2147483647. mstime_t is long long https://github.com/redis/redis/blob/f11a2d4dd764c996b2d0c0cb5abde13f2445b40c/src/server.h#L58.
When the value is sentinel_election_timeout is larger than 2147483647 election_timeout is set to sentinel_election_timeout then election_timeout value will be negative. So setting election_timeout to mstime_t makes sense to avoid the potential bug and to be consistent.
But I am not sure in what case the value of sentinel_election_timeout will be set to a value larger than 2147483647.

Reply:
This bug will cause the failover fail and keep getting delayed indefinitely. If sentinel_election_timeout is larger than 2147483647 then election_timeout will be negative.
Then the following condition will be true and the failover will be aborted.
https://github.com/redis/redis/blob/f27083a4a8a6682e391a533724c904c69852c0a0/src/sentinel.c#L4923

Reply:
According to Yossi it'll only happen when the user input is broken, and in that case it'll be immediately visible.
so not something to fuss about.